src/components/Common/SearchBar.tsx

```diff
@@ -1,25 +1,211 @@
 import SearchRoundedIcon from "@mui/icons-material/SearchRounded";
-import { InputAdornment, TextField } from "@mui/material";
+import ClearIcon from "@mui/icons-material/Clear";
+import {
+  Autocomplete,
+  Avatar,
+  Box,
+  IconButton,
+  InputAdornment,
+  ListItemAvatar,
+  ListItemButton,
+  ListItemText,
+  TextField,
+  Typography,
+  debounce,
+} from "@mui/material";
+import { useEffect, useMemo, useRef, useState } from "react";
+import axios from "axios";
+import { useAuth0 } from "@auth0/auth0-react";
+import { SelectedUser } from "../../state/slices/messagesSlice";
+import { useNavigate } from "react-router-dom";
 
 type SearchBarProps = {
   placeholder: string;
 };
 
+const styles = {
+  autocomplete: {
+    "&.MuiAutocomplete-input": { paddingX: 0 },
+    position: "relative",
+  },
+  box: {
+    paddingBottom: 1,
+    paddingTop: 0,
+    paddingX: 2,
+  },
+  displayName: {
+    fontWeight: "bold",
+  },
+  listBox: {
+    maxHeight: "60vh",
+    overflowY: "auto",
+    zIndex: 1300,
+  },
+  searchIcon: {
+    paddingRight: 0,
+    "&.Mui-disabled": {
+      color: "gray.main",
+    },
+  },
+  searchIconFocused: {
+    paddingRight: 0,
+    "&.Mui-disabled": {
+      color: "primary.main",
+    },
+  },
+};
+
 const SearchBar = ({ placeholder }: SearchBarProps) => {
+  const debouncedSetter = useMemo(
+    () => debounce((keywords: string) => setKeywords(keywords), 20),
+    []
+  );
+  const [focusSearchBar, setFocusSearchBar] = useState(false);
+  const inputRef = useRef<HTMLInputElement>(null);
+  const [keywords, setKeywords] = useState("");
+  const [loading, setLoading] = useState(false);
+  const [userList, setUserList] = useState<SelectedUser[]>([]);
+  const { getAccessTokenSilently } = useAuth0();
+  const navigate = useNavigate();
+
+  const fetchUsers = async () => {
+    setLoading(true);
+    if (keywords.length === 0) return {};
+    try {
+      const token = await getAccessTokenSilently();
+      const result = await axios.get(
+        `http://localhost:3001/api/users/getUsers`,
+        {
+          params: {
+            keyword: keywords,
+          },
+          headers: {
+            Authorization: `Bearer ${token}`,
+          },
+        }
+      );
+
+      setUserList(result.data as SelectedUser[]);
+
+      return result.data;
+    } catch (error) {
+      console.error("Failed to fetch users:", error);
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  const handleClear = () => {
+    setKeywords("");
+    setUserList([]);
+  };
+
+  const handleInputChange = (newInputValue: string) => {
+    debouncedSetter(newInputValue);
+    setFocusSearchBar(true);
+    if (newInputValue === "") {
+      setUserList([]);
+    }
+  };
+
+  const onSelect = (selectedUsername: string) => {
```

Comment:
We should set the `userList` to an empty array here so that when someone selects something the options array is empty